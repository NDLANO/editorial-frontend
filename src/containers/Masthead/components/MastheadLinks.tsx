/**
 * Copyright (c) 2024-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { ComponentPropsWithoutRef } from "react";
import { SafeLink } from "@ndla/safelink";
import { styled } from "@ndla/styled-system/jsx";
import { JsxStyleProps } from "@ndla/styled-system/types";

const LinksContainer = styled("div", {
  base: {
    display: "flex",
    gap: "xsmall",
  },
});

export const MastheadLinks = (props: ComponentPropsWithoutRef<"div"> & JsxStyleProps) => {
  return (
    <LinksContainer {...props}>
      <SafeLink target="_blank" to="https://edndla.zendesk.com/hc/no">
        Zendesk
      </SafeLink>
      <SafeLink target="_blank" to="https://kvalitet.ndla.no/">
        Kvalitaisen
      </SafeLink>
    </LinksContainer>
  );
};
