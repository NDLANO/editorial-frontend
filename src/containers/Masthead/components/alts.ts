/**
 * Copyright (c) 2025-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

const alts = {
  nb: [
    "Norske Dovendyrs Laug og Allianse",
    "Nyttige Dagens Lunsj Anbefalinger",
    "Nasjonal Dugnad for Latter og Anarki",
    "Nysgjerrige Detektivers Løsnings Arkiv",
    "Nordmenns Daglige Lykke Aktivitet",
    "Nattens Dansende Lyskastere Allianse",
    "Nyttige Data for Lærings Aktiviteter",
    "Norges Dyreste Luksus Artikler",
    "Nydelige Dyr Lager Abstraksjoner",
    "Nasjonal Dugnad for Løft av Alpakkaer",
  ],
  nn: [
    "Norske Dovendyrs Laug og Allianse",
    "Nyttige Dagens Lunsj Anbefalinger",
    "Nasjonal Dugnad for Latter og Anarki",
    "Nysgjerrige Detektivers Løsnings Arkiv",
    "Nordmenns Daglige Lykke Aktivitet",
    "Nattens Dansende Lyskastere Allianse",
    "Nyttige Data for Lærings Aktiviteter",
    "Norges Dyreste Luksus Artikler",
    "Nydelige Dyr Lager Abstraksjoner",
    "Nasjonal Dugnad for Løft av Alpakkaer",
  ],
  en: [
    "Never Drink Lemonade Alone",
    "Nocturnal Dancing Llama Association",
    "Ninja Donuts Love Adventures",
    "New Dinosaur Language Academy",
    "Notorious Duck Lovers Anonymous",
    "National Day of Lazy Activities",
    "Nerds Discussing Lizard Anatomy",
    "Neverending Dreams of Lush Adventures",
    "Nomadic Dragons Learning Algebra",
    "Nostalgic Doodles and Laughs Assembly",
  ],
};
export default alts;
