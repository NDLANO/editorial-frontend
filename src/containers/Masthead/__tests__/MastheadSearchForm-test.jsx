/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import nock from 'nock';
import { render } from '@testing-library/react';
import { MastheadSearchForm } from '../components/MastheadSearchForm';
import { taxonomyApi } from '../../../config';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';

const noop = () => {};

test('MastheadSearchForm redirects on ndla url paste with id at the end', () => {
  const historyMock = {
    push: jest.fn(),
  };

  const { container } = render(
    <IntlWrapper>
      <MastheadSearchForm
        query=""
        searching={false}
        onSearchQuerySubmit={noop}
        t={() => ''}
        locale="nb"
        history={historyMock}
      />
    </IntlWrapper>,
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push.calledOnce).toBeTruthy();
    expect(historyMock.push.calledWith('/subject-matter/learning-resource/3333/edit/nb')).toBe(
      true,
    );
  }, global.DEFAULT_TIMEOUT);
});

test('MastheadSearchForm redirects on ndla url paste with taxonomy id at the end', () => {
  const historyMock = {
    push: jest.fn(),
  };

  nock('http://ndla-api')
    .get(`${taxonomyApi}/topics/urn:topic:1:179373/?language=nb`)
    .reply(200, { contentUri: 'urn:content:4232' });

  const { container } = render(
    <MastheadSearchForm
      query="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373"
      searching={false}
      locale="nb"
      onSearchQuerySubmit={noop}
      t={() => ''}
      history={historyMock}
    />,
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push.calledOnce).toBeTruthy();
    expect(historyMock.push.calledWith('/subject-matter/topic-article/4232/edit/nb')).toBe(true);
  }, global.DEFAULT_TIMEOUT);
});

test('MastheadSearchForm redirects on old ndla url paste with new id', () => {
  const historyMock = {
    push: jest.fn(),
  };

  nock('http://ndla-api')
    .get('/draft-api/v1/drafts/external_id/4737')
    .reply(200, { id: '123' });

  const { container } = render(
    <MastheadSearchForm
      query="https://ndla.no/nb/node/4737?fag=36"
      searching={false}
      locale="nb"
      onSearchQuerySubmit={noop}
      t={() => ''}
      history={historyMock}
    />,
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push.calledOnce).toBeTruthy();
    expect(historyMock.push.getCall(0).args[0]).toBe(
      '/subject-matter/learning-resource/123/edit/nb',
    );
  }, global.DEFAULT_TIMEOUT);
});

test('MastheadSearchForm invalid id at the end of the url', () => {
  const historyMock = {
    push: jest.fn(),
  };

  const { container } = render(
    <MastheadSearchForm
      query="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373/urn:resource:1:16838"
      searching={false}
      onSearchQuerySubmit={noop}
      t={() => ''}
      locale="nb"
      history={historyMock}
    />,
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push.calledOnce).toBe(false);
  }, global.DEFAULT_TIMEOUT);
});

test('MastheadSearchForm redirects on ndla node id pasted', () => {
  const historyMock = {
    push: jest.fn(),
  };
  nock('http://ndla-api')
    .get('/draft-api/v1/drafts/external_id/4737')
    .reply(200, { id: '123' });

  const { container } = render(
    <MastheadSearchForm
      query="#4737"
      searching={false}
      locale="nb"
      onSearchQuerySubmit={noop}
      t={() => ''}
      history={historyMock}
    />,
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push.calledOnce).toBeTruthy();
    expect(historyMock.push.calledWith('/subject-matter/learning-resource/123/edit/nb')).toBe(true);
  }, global.DEFAULT_TIMEOUT);
});
