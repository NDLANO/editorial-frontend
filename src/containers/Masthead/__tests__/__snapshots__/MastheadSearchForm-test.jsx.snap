// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MastheadSearchForm invalid id at the end of the url 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6.5px;
  height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0[lang='zh'],
.emotion-0[lang='zh-Hans'],
.emotion-0[lang='zh-Hant'] {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6.5px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6.5px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4.333333333333333px 13px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1[lang='zh'],
.emotion-1[lang='zh-Hans'],
.emotion-1[lang='zh-Hant'] {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 13px;
  width: 24px;
  height: 24px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="c-icon emotion-2"
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align: middle; color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
        width="1em"
      >
        <g>
          <path
            d="M31 28h-1.59l-.55-.55C30.82 25.18 32 22.23 32 19c0-7.18-5.82-13-13-13S6 11.82 6 19s5.82 13 13 13c3.23 0 6.18-1.18 8.45-3.13l.55.55V31l10 9.98L40.98 38 31 28zm-12 0c-4.97 0-9-4.03-9-9s4.03-9 9-9 9 4.03 9 9-4.03 9-9 9z"
          />
        </g>
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373/urn:resource:1:16838"
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on ndla node id pasted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6.5px;
  height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0[lang='zh'],
.emotion-0[lang='zh-Hans'],
.emotion-0[lang='zh-Hant'] {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6.5px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6.5px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4.333333333333333px 13px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1[lang='zh'],
.emotion-1[lang='zh-Hans'],
.emotion-1[lang='zh-Hant'] {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 13px;
  width: 24px;
  height: 24px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="c-icon emotion-2"
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align: middle; color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
        width="1em"
      >
        <g>
          <path
            d="M31 28h-1.59l-.55-.55C30.82 25.18 32 22.23 32 19c0-7.18-5.82-13-13-13S6 11.82 6 19s5.82 13 13 13c3.23 0 6.18-1.18 8.45-3.13l.55.55V31l10 9.98L40.98 38 31 28zm-12 0c-4.97 0-9-4.03-9-9s4.03-9 9-9 9 4.03 9 9-4.03 9-9 9z"
          />
        </g>
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="#4737"
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on ndla url paste with id at the end 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6.5px;
  height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0[lang='zh'],
.emotion-0[lang='zh-Hans'],
.emotion-0[lang='zh-Hant'] {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6.5px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6.5px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4.333333333333333px 13px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1[lang='zh'],
.emotion-1[lang='zh-Hans'],
.emotion-1[lang='zh-Hant'] {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 13px;
  width: 24px;
  height: 24px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="c-icon emotion-2"
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align: middle; color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
        width="1em"
      >
        <g>
          <path
            d="M31 28h-1.59l-.55-.55C30.82 25.18 32 22.23 32 19c0-7.18-5.82-13-13-13S6 11.82 6 19s5.82 13 13 13c3.23 0 6.18-1.18 8.45-3.13l.55.55V31l10 9.98L40.98 38 31 28zm-12 0c-4.97 0-9-4.03-9-9s4.03-9 9-9 9 4.03 9 9-4.03 9-9 9z"
          />
        </g>
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on ndla url paste with taxonomy id at the end 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6.5px;
  height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0[lang='zh'],
.emotion-0[lang='zh-Hans'],
.emotion-0[lang='zh-Hant'] {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6.5px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6.5px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4.333333333333333px 13px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1[lang='zh'],
.emotion-1[lang='zh-Hans'],
.emotion-1[lang='zh-Hant'] {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 13px;
  width: 24px;
  height: 24px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="c-icon emotion-2"
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align: middle; color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
        width="1em"
      >
        <g>
          <path
            d="M31 28h-1.59l-.55-.55C30.82 25.18 32 22.23 32 19c0-7.18-5.82-13-13-13S6 11.82 6 19s5.82 13 13 13c3.23 0 6.18-1.18 8.45-3.13l.55.55V31l10 9.98L40.98 38 31 28zm-12 0c-4.97 0-9-4.03-9-9s4.03-9 9-9 9 4.03 9 9-4.03 9-9 9z"
          />
        </g>
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373"
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on old ndla url paste with new id 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6.5px;
  height: 52px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0[lang='zh'],
.emotion-0[lang='zh-Hans'],
.emotion-0[lang='zh-Hant'] {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6.5px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6.5px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4.333333333333333px 13px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.625;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1[lang='zh'],
.emotion-1[lang='zh-Hans'],
.emotion-1[lang='zh-Hant'] {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 13px;
  width: 24px;
  height: 24px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="c-icon emotion-2"
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align: middle; color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
        width="1em"
      >
        <g>
          <path
            d="M31 28h-1.59l-.55-.55C30.82 25.18 32 22.23 32 19c0-7.18-5.82-13-13-13S6 11.82 6 19s5.82 13 13 13c3.23 0 6.18-1.18 8.45-3.13l.55.55V31l10 9.98L40.98 38 31 28zm-12 0c-4.97 0-9-4.03-9-9s4.03-9 9-9 9 4.03 9 9-4.03 9-9 9z"
          />
        </g>
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="https://ndla.no/nb/node/4737?fag=36"
    />
  </form>
</div>
`;
